server/src/controllers/userController.js

```diff
@@ -22,6 +23,34 @@ const createNewUser = async (req, res) => {
   }
 };
 
+const loginUser = async (req, res) => {
+  const { email, password } = req.body;
+
+  try {
+    const user = await User.findOne({ email });
```

Comment:
```suggestion
    const user = await User.findOne({ email });
    
    if (!user) {
        return res.status(400).send({
           errors: 'No user record found',
        });
    }

    const passwordMatch = await bcrypt.compare(password, user.password);

     if (!passwordMatch) {
        return res.status(401).send({
          errors: 'Invalid password',
         });
      }
      
      const token = user.generateAuthToken();
      res.set('X-Auth-Token', token);
      return res.send(user);
      
```

Reply:
@nimish-gupta Good catch. I will make this change.

Reply:
@nimish-gupta The changes have been made.